avm/utilities/pipelines/e2eValidation/resourceDeployment/New-TemplateDeployment.ps1

```diff
@@ -164,20 +165,19 @@ function New-TemplateDeploymentInner {
             $deploymentNamePrefix = 'templateDeployment-{0}' -f (Split-Path $templateFilePath -LeafBase)
         }
 
-        $modulesRegex = '.+[\\|\/]modules[\\|\/]'
-        if ($templateFilePath -match $modulesRegex) {
-            # If we can assume we're operating in a module structure, we can further fetch the provider namespace & resource type
-            $shortPathElem = (($templateFilePath -split $modulesRegex)[1] -replace '\\', '/') -split '/' # e.g., app-configuration, configuration-store, .test, common, main.test.bicep
-            $providerNamespace = $shortPathElem[0] # e.g., app-configuration
-            $providerNamespaceShort = ($providerNamespace -split '-' | ForEach-Object { $_[0] }) -join '' # e.g., ac
-
-            $resourceType = $shortPathElem[1] # e.g., configuration-store
-            $resourceTypeShort = ($resourceType -split '-' | ForEach-Object { $_[0] }) -join '' # e.g. cs
-
-            $testFolderShort = Split-Path (Split-Path $templateFilePath -Parent) -Leaf  # e.g., common
-
-            $deploymentNamePrefix = "$providerNamespaceShort-$resourceTypeShort-$testFolderShort" # e.g., ac-cs-common
+        # Convert, e.g., [C:\myFork\avm\res\kubernetes-configuration\flux-configuration\tests\e2e\defaults\main.test.bicep] to [a-r-kc-fc-defaults]
+        $shortPathElems = ((Split-Path $templateFilePath) -replace ('{0}[\\|\/]' -f [regex]::Escape($repoRoot))) -split '[\\|\/]' | Where-Object { $_ -notin @('tests', 'e2e') }
```

Comment:
Since deployment names have lenght constraints, should we trim everything but the unique suffix (date) in case it exceeds the limit?

Reply:
The trim is built-in below :)